Source/WebKit/WebProcess/WebCoreSupport/WebPermissionController.cpp

```diff
@@ -26,33 +26,33 @@
 #include "config.h"
 #include "WebPermissionController.h"
 
-
 #include "WebPage.h"
+#include "WebPageProxyIdentifier.h"
 #include "WebPageProxyMessages.h"
+#include "WebPermissionControllerProxyMessages.h"
+#include "WebProcess.h"
+#include <WebCore/Page.h>
 #include <WebCore/PermissionObserver.h>
 
 namespace WebKit {
 
-Ref<WebPermissionController> WebPermissionController::create(WebPage& page)
+Ref<WebPermissionController> WebPermissionController::create()
 {
-    return adoptRef(*new WebPermissionController(page));
+    return adoptRef(*new WebPermissionController());
 }
 
-WebPermissionController::WebPermissionController(WebPage& page)
-    : m_page(page)
+WebPermissionController::WebPermissionController()
 {
 }
 
-void WebPermissionController::query(WebCore::ClientOrigin&& origin, WebCore::PermissionDescriptor&& descriptor, CompletionHandler<void(std::optional<WebCore::PermissionState>)>&& completionHandler)
+void WebPermissionController::query(WebCore::ClientOrigin&& origin, WebCore::PermissionDescriptor&& descriptor, WebCore::Page& page, CompletionHandler<void(std::optional<WebCore::PermissionState>)>&& completionHandler)
 {
-    if (!m_page)
-        return completionHandler({ });
-
     auto cachedResult = queryCache(origin, descriptor);
     if (cachedResult != WebCore::PermissionState::Prompt)
         return completionHandler(cachedResult);
 
-    m_requests.append(PermissionRequest { WTFMove(origin), WTFMove(descriptor), WTFMove(completionHandler) });
+    auto webPageProxyIdentifier = WebPage::fromCorePage(page).webPageProxyIdentifier();
+    m_requests.append(PermissionRequest { WTFMove(origin), WTFMove(descriptor), WTFMove(completionHandler), WTFMove(webPageProxyIdentifier) });
```

Comment:
I think it'd look nicer if the completion handler came last.

Reply:
No WTFMove() for webPageProxyIdentifier since this is a uint64_t.

Reply:
Fixed!